designer/server/src/common/helpers/build-error-details.js

```diff
@@ -1,20 +1,36 @@
 /**
- * @param {ValidationErrorItem[]} errorDetails
+ * @param {ValidationError} error
  */
-export function buildErrorDetails(errorDetails) {
-  return errorDetails.reduce((errors, detail) => {
+export function buildErrorDetails(error) {
+  return error.details.reduce((errors, { context, message }) => {
+    if (!context?.key) {
+      return errors
+    }
+
     return {
-      [detail.context?.key ?? 'unknown']: {
-        message: detail.message
+      [context.key]: {
+        text: message,
+        href: `#${context.key}`
```

Comment:
I've changed the error `{ message }` to `{ text, href }`

This keeps our errors aligned with the [**Error summary** component](https://design-system.service.gov.uk/components/error-summary/#options-error-summary-example--error-list)